athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneMetadataHandler.java

```diff
@@ -0,0 +1,249 @@
+        Schema tableSchema = null;
+        try {
+            if (glue != null) {
+                tableSchema = super.doGetTable(blockAllocator, request).getSchema();
+                logger.info("doGetTable: Retrieved schema for table[{}] from AWS Glue.", request.getTableName());
+            }
+        } catch (RuntimeException ex) {
+            logger.warn("doGetTable: Unable to retrieve table[{}:{}] from AWS Glue.",
+                    request.getTableName().getSchemaName(), request.getTableName().getTableName(), ex);
+        }
+        return new GetTableResponse(request.getCatalogName(), request.getTableName(), tableSchema);
+    }
+
+    /**
+     * Our table doesn't support complex layouts or partitioning so we simply make
+     * this method a NoOp.
+     */
+    @Override
+    public void getPartitions(BlockWriter blockWriter, GetTableLayoutRequest request,
+            QueryStatusChecker queryStatusChecker) throws Exception {
+
+        // No implemenation as connector doesn't support partitioning
+    }
+
+    /**
+     * Used to split-up the reads required to scan the requested batch of
+     * partition(s).
+     *
+     * @param blockAllocator Tool for creating and managing Apache Arrow Blocks.
+     * @param request        Provides details of the catalog, database, table,
+     *                       andpartition(s) being queried as well as any filter
+     *                       predicate.
+     * @return A GetSplitsResponse which primarily contains: 1. A Set<Split> which
+     *         represent read operations Amazon Athena must perform by calling your
+     *         read function. 2. (Optional) A continuation token which allows you to
+     *         paginate the generation of splits for large queries.
+     * @note A Split is a mostly opaque object to Amazon Athena. Amazon Athena will
+     *       use the optional SpillLocation and optional EncryptionKey for pipelined
+     *       reads but all properties you set on the Split are passed to your read
+     *       function to help you perform the read.
+     */
+
+    /*
+     * Since our connector does not support parallel scans we generate a single
+     * Split and include the connection details as a property on the split so that
+     * the RecordHandler has easy access to it.
+     */
+    @Override
+    public GetSplitsResponse doGetSplits(BlockAllocator blockAllocator, GetSplitsRequest request) {
+        // Every split must have a unique location if we wish to spill to avoid failures
+        SpillLocation spillLocation = makeSpillLocation(request);
+
+        // Since our connector does not support parallel reads we return a fixed split.
+        return new GetSplitsResponse(request.getCatalogName(),
+                Split.newBuilder(spillLocation, makeEncryptionKey()).build());
+    }
+
+    @Override
+    protected Field convertField(String name, String glueType) {
+        return GlueFieldLexer.lex(name, glueType);
```

Comment:
Where/how are you handling type conversion from Glue to Neptune? I've only seen code for Int(32) and utf8 so far.

Reply:
GremlinQueryPreProcessor is responsible for identifying column type and pickup correct query templates. There are only few distinct neptune where clause variations which change across types. I have currently put in code for Int and String. for eg: .has('<key>',eq('<value>')) and has('<key>',eq(<value>) the only difference is the single quotes. I have kept these as separate to do a cleaner implementation. I need to add code to intercept more data types and apply appropriate templates